mm2src/coins/lightning/ln_events.rs

```diff
@@ -284,6 +456,50 @@ impl LightningEventHandler {
                 return;
             },
         };
-        platform_coin.broadcast_transaction(&spending_tx);
+
+        self.platform.broadcast_transaction(&spending_tx);
+
+        let claiming_tx_inputs_value = outputs.iter().fold(0, |sum, output| match output {
+            SpendableOutputDescriptor::StaticOutput { output, .. } => sum + output.value,
+            SpendableOutputDescriptor::DelayedPaymentOutput(descriptor) => sum + descriptor.output.value,
+            SpendableOutputDescriptor::StaticPaymentOutput(descriptor) => sum + descriptor.output.value,
+        });
+        let claiming_tx_outputs_value = spending_tx.output.iter().fold(0, |sum, txout| sum + txout.value);
+        if claiming_tx_inputs_value < claiming_tx_outputs_value {
+            log::error!(
+                "Claiming transaction input value {} can't be less than outputs value {}!",
+                claiming_tx_inputs_value,
+                claiming_tx_outputs_value
+            );
+            return;
+        }
+        let claiming_tx_fee = claiming_tx_inputs_value - claiming_tx_outputs_value;
+        let claiming_tx_fee_per_channel = (claiming_tx_fee as f64) / (outputs.len() as f64);
+
+        for output in outputs {
+            let (closing_txid, claimed_balance) = match output {
+                SpendableOutputDescriptor::StaticOutput { outpoint, output } => {
+                    (outpoint.txid.to_string(), output.value)
+                },
+                SpendableOutputDescriptor::DelayedPaymentOutput(descriptor) => {
```

Comment:
Since the action of this patterns are the same, you can avoid code duplication by merging 2 patterns.

Example:
```rust
SpendableOutputDescriptor::DelayedPaymentOutput(descriptor) | SpendableOutputDescriptor::StaticPaymentOutput(descriptor) => {
    (descriptor.outpoint.txid.to_string(), descriptor.output.value)
}

```

Reply:
Unfortunately, this can't be done as `descriptor` is of a different type in every pattern.

Reply:
> Unfortunately, this can't be done as `descriptor` is of a different type in every pattern.

Ah, didn't know that. Sorry.